### PR TITLE
feature/PRO-1037/include-fees-uncollected-positions

### PR DESCRIPTION
**Pull request by Janislav**
# Pull Request

Closes: PRO-1037

## Checklist

Please conduct a thorough self-review before opening the PR.

- [ ] I am confident that the code works.
- [ ] I have updated documentation where appropriate.

## Summary

*Please include a succinct description of the purpose and content of the PR. What problem does it solve, and how? Link issues, discussions, other PRs, and anything else that will help the reviewer.*

### CODE REVIEW

File: bouncer/shared/lp_api_test.ts
Comment (Janislav) [inline bouncer/shared/lp_api_test.ts:50]: Like it seemed this test has never tested anything, ethBalance was undefined and undefined < a number is false :D. Thanks JavaScript :D.

File: api/bin/chainflip-lp-api/src/main.rs
Comment (AlastairHolmes) [inline api/bin/chainflip-lp-api/src/main.rs:309]: `result` is a bit ambiguous, maybe `asset_balances`

**Review by martin-chainflip — APPROVED** [review]

**Comment by martin-chainflip** [comment]
Please don't merge until Alastair has approved.

File: state-chain/runtime/src/runtime_apis.rs
Comment (AlastairHolmes) [inline state-chain/runtime/src/runtime_apis.rs:59]: I assume you need this type, as RpcAsset isn't no-std compatible? I believe we could remove it in the future, and have added a note to PRO-1157 to do so.

File: state-chain/runtime/src/lib.rs
Comment (AlastairHolmes) [inline state-chain/runtime/src/lib.rs:1025]: Don't think this conversion is needed, yo seem to be going from ChainAsset to ChainAsset.

File: state-chain/pallets/cf-lp/src/lib.rs
Comment (AlastairHolmes) [inline state-chain/pallets/cf-lp/src/lib.rs:368]: I think it probably makes sense to delete this AssetBalance type, and just use a tuple of ChainAsset, and AssetBalance.

Comment (AlastairHolmes) [reply state-chain/pallets/cf-lp/src/lib.rs:368]: Basically so it matches the interface of: cf_asset_balances

**Review by AlastairHolmes — APPROVED** [review]
Just this one change to remove the new `struct AssetBalance`, then happy to merge. Thank you.

**Comment by AlastairHolmes** [comment]
You could instead keep the AssetBalances struct, but change the cf_asset_balances rpc to return it.

Currently cf_asset_balances would return something like this: `[{"chain":"Bitcoin","asset":"BTC"},34]`

If we keep the cf_asset_balances struct we could have the encoding look like this (via serde(flatten)):
{"chain":"Bitcoin","asset":"BTC", "amount": 34}

File: state-chain/pallets/cf-lp/src/lib.rs
Comment (Janislav) [reply state-chain/pallets/cf-lp/src/lib.rs:368]: I think we can't because it's returned from the lp-api, and I'm pretty sure we don't want to have breaking changes, right?

File: state-chain/runtime/src/runtime_apis.rs
Comment (Janislav) [reply state-chain/runtime/src/runtime_apis.rs:59]: In theory, we could move it to the runtime, but it's a type created for the RPC and I don't think it belongs there. Importing it from the RPC to the runtime is not possible because the dependencies are already set the other way around. But in any case we need a type that reflects the chain and the asset also for the custom RPC of the state-chain to the outer world. Internally, we can cast from Asset to Chain, but that's not possible if we return it from RPC. I think the cause of this type just shows that we have a general problem with our type system. The whole thing breaks at the moment with introduce USDC (for e.x) on any other chain.

File: state-chain/pallets/cf-lp/src/lib.rs
Comment (AlastairHolmes) [reply state-chain/pallets/cf-lp/src/lib.rs:368]: There are two types both called AssetBalance, I mean the other one.

Comment (AlastairHolmes) [reply state-chain/pallets/cf-lp/src/lib.rs:368]: But yes we don't want breaking changes.

File: state-chain/runtime/src/runtime_apis.rs
Comment (AlastairHolmes) [inline state-chain/runtime/src/runtime_apis.rs:60]: I'm not sure this will do anything in this case, same with the asset below.

Comment (AlastairHolmes) [reply state-chain/runtime/src/runtime_apis.rs:60]: I think this type isn't used now?

Comment (Janislav) [reply state-chain/runtime/src/runtime_apis.rs]: Good catch 🕵️

**Review by AlastairHolmes — APPROVED** [review]

**Comment by AlastairHolmes** [comment]
Good to merge

**Pull request merged**
